### PR TITLE
Feature/render features

### PR DESCRIPTION
**Pull request by gocarlos**
This is only a first approach to the rendering: 

Please have a look on how I organized the code: 
* gl commands are in [renderer.cc ](https://github.com/OpenMapper/OpenMapper/compare/fd52562ca0031cbf70fec4595c1100083ac7f9fc...15edbf03ca634857c31cd720816f0e9e43e25d94#diff-ea4941fc9f8bbd4247fb786cc6d9b264R32) the engine code and they are called from the GUI classes like the implemented Qt in myGLWidget.cc in this Desktop app. 

Does this make sense for you? would that work later with iOS?

### CODE REVIEW

**Comment by gocarlos** [comment]
sorry for all the fix commits, did some late work with the ipad, and had a lot of typos :dancer:

File: include/openmapper_desktop/config.h [hunk 1]
```diff
@@ -5,9 +5,10 @@
 
 #include <string>
 
-std::string path_to_vocabulary =
-    "libs/OpenMapper/thirdparty/slam_engine/ORB_SLAM2/Vocabulary/ORBvoc.txt";
-std::string path_to_settings =
-    "libs/OpenMapper/thirdparty/slam_engine/ORB_SLAM2/Vocabulary/webcam.yaml";
+// OpenMapper
```
Comment (gocarlos) [inline include/openmapper_desktop/config.h:8]: This was removed as we implemented that in the engine.

File: include/openmapper_desktop/window.h [hunk 2]
```diff
@@ -25,25 +23,26 @@ class Window : public QWidget {
...
  protected:
-  void keyPressEvent(QKeyEvent *event);
-  void timerEvent(QTimerEvent *event);
+  void keyPressEvent(QKeyEvent* event);
+  void timerEvent(QTimerEvent* event);
   void initialize_input();
 
  private:
-  Ui::Window *ui;
+  Ui::Window* ui;
```
Comment (gocarlos) [inline include/openmapper_desktop/window.h]: Dont know right now why all pointer signs are left instead of right. Will correct that later

**Comment by gocarlos** [comment]
As far as I know the whole code used right know is compatible with OpenGL ES :dancing_men:

**Review by nidegen — APPROVED** [review]
looks awesome. do you code on the ipad?!?:)

i think long term, some rendering code may even be part of the core OpenMapper. not sure how much we can share between ios gles and qt right now though

**Comment by nidegen** [comment]
ah ok i now see that part of it is in the main openmapper repo:) very nice

**Comment by gocarlos** [comment]
yes, I thought the whole rendering should be in the openmapper. by the way the gl used inside qt is gles compatible as qt also targets android :)..

regarding coding with the ipad -> just when I cannot sleep :P

**Pull request merged**